src/Security/SecProtocolMetadata.cs

```diff
@@ -51,9 +51,177 @@ public class SecProtocolMetadata : NativeObject {
 
 		public bool EarlyDataAccepted => sec_protocol_metadata_get_early_data_accepted (GetCheckedHandle ()) != 0;
 
-		//
-		// MISSING: all the block APIs
-		//
+ 		[DllImport (Constants.SecurityLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
+		extern static bool sec_protocol_metadata_challenge_parameters_are_equal (IntPtr metadataA, IntPtr metadataB);
 
+		public static bool ChallengeParametersAreEqual (SecProtocolMetadata metadataA, SecProtocolMetadata metadataB)
+		{
+			if (metadataA == null && metadataB == null)
+				return false;  // This was tested in a native app. We do copy the behaviour.
+			if (metadataA == null)
+				throw new ArgumentNullException (nameof (metadataA));
+			if (metadataB == null)
+				throw new ArgumentNullException (nameof (metadataB));
```

Comment:
does the API allows `nil` ? the previous comment makes it sound like it does,  if that's the case then we should too, so no throwing `ANE`. The common pattern is:
```
if (a == null)
   return (b == null);
else if (b == null)
   return false;
return sec_...(a, b);
```

Reply:
or, if `nil` is allowed then it might be better to use `GetHandle` extension method, to avoid any exception to be thrown (not really needed since the decision should be based on the `bool` result), e.g.
```
return sec_protocol_metadata_challenge_parameters_are_equal (metadataA.GetHandle (), metadataB.GetHandle ());
```

Reply:
@spouliot the API does allow the null, but a warning is raised (as in passing NULL, NULL) will try with a one null and the other one present.

Reply:
If the headers don't mention null is fine and the OS shows warning in the logs then let's assume null is not fine (at least to give to the API). In that case the first approach (exception less) is better than the `GetHandle` one.

Reply:
:+1: